test-suite/run_djinni.sh

```diff
@@ -73,6 +75,35 @@ fi
     --cpp-optional-template "std::experimental::optional" \
     --cpp-optional-header "<experimental/optional>" \
     --cpp-extended-record-include-prefix "../../handwritten-src/cpp/" \
+    --cpp-use-wide-strings true \
```

Comment:
Can you re-structure this and add some comments to make it clear that Djinni is being run twice, and why that is the case?  There are a few tricky details here, like the fact that you're letting only the second command fill in the list of infiles/outfiles.  That works for the test suite, though it wouldn't work if those were actually being used to detect dependency changes.

We desperately need a higher-level structure of the test suite which allows for multiple Djinni runs as separate test cases.  Until that's developed, this is a reasonable approach, and works since there's no sharing between your wchar_test.djinni and the main all.djinni file.